### PR TITLE
Table processing

### PR DESCRIPTION
**Pull request by gumaerc**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-to-hugo/issues/10

#### What's this PR do?
Adds a turndown rule to process content in raw HTML tables brought in from OCW course sections so they are rendered properly by Hugo.

#### How should this be manually tested?
Run any number of courses through this tool, then take the markdown output and load it into hugo-course-publisher.  The assignments or readings sections usually have tables, and they should now display properly.

#### Screenshots (if appropriate)
image

### CODE REVIEW

File: src/markdown_generators.js [hunk 1]
```diff
@@ -4,35 +4,66 @@ const yaml = require("js-yaml")
...
+     * from earlier with the HTML character entity for a pipe
+     * reintroduce them between rows, replacing our pipe marker
+     */
+    content = content
+      .substring(content.indexOf("|"), content.lastIndexOf("|"))
+      .replace(/\r?\n|\r/g, "{{< br >}}")
+      .replace(/\|{{< br >}}\|/g, "|\n|")
+      .replace(/REPLACETHISWITHAPIPE/g, "&#124;")
+    // Only do this if header lines are found
+    if (content.match(/(?<=\| \*\*)(.*?)(?=\*\* \|\r?\n|\r)/g)) {
```
Comment (aliceriot) [inline src/markdown_generators.js]: some of these longer / more complicated regexes it might be a good idea to pull out as a separate viable and just writing a little comment describing what they're supposed to match

File: src/markdown_generators.js [hunk 2]
```diff
@@ -4,35 +4,66 @@ const yaml = require("js-yaml")
...
+     */
+    for (let i = 0; i < node.rows.length; i++) {
+      const cells = node.rows[i].cells
+      for (let j = 0; j < cells.length; j++) {
+        cells[j].innerHTML = cells[j].innerHTML.replace(
+          /\|/g,
+          REPLACETHISWITHAPIPE
+        )
+      }
+    }
```
[the remark this answers is not in the document]
Comment (gumaerc) [reply src/markdown_generators.js]: I avoided doing it this way specifically because the "node" object passed into this function is a native DOM HTMLElement.  The code actually breaks if I put in what you have above, because forEach isn't a function on `.rows` or `.cells`.

File: src/markdown_generators.js [hunk 3]
```diff
@@ -82,8 +82,8 @@ turndownService.addRule("refshortcode", {
     const ref = turndownService.escape(
       node
         .getAttribute("href")
-        .replace(REFSHORTCODESTART, '{{< ref "')
-        .replace(REFSHORTCODEEND, '" >}}')
+        .replace(REFSHORTCODESTART, '{{% ref "')
+        .replace(REFSHORTCODEEND, '" %}}')
```
Comment (aliceriot) [inline src/markdown_generators.js]: just curious, why was this change necessary?

File: src/markdown_generators.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (aliceriot) [reply src/markdown_generators.js]: ahh yeah on HTMLElement you have to do something like [...el].forEach to be able to use it.. tbh I don't know why .forEach isn't implemented there...kind of annoying.

File: src/markdown_generators.js (hunk 3 above)
Comment (gumaerc) [reply src/markdown_generators.js]: It wasn't actually necessary per se, but:

> In Hugo 0.55 we changed how the % delimiter works. Shortcodes using the % as the outer-most delimiter will now be fully rendered when sent to the content renderer

I figured we should standardize on using shortcodes this way, even if it doesn't really matter for the `ref` shortcode.  This means that the output of the shortcode will be rendered as markdown fully before the final markdown processing step.
